File: GymNotes.Entity/Models/NewFolder/TrainingPlan.cs
using GymNotes.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace GymNotes.Entity.Models.NewFolder
{
  public class TrainingPlan
  {
    [Key]
    public int Id { get; set; }

    [Required]
    public string CreatorId { get; set; }

    [Required]
    public virtual ApplicationUser Creator { get; set; }

    [Required]
    public string OwnerId { get; set; }

    [Required]
    public virtual ApplicationUser Owner { get; set; }

    [Required]
    public DateTime ModifiedAt { get; set; }

    [Required]
    public string Description { get; set; }

    [Required]
    public string Name { get; set; }

    public virtual ICollection<TrainingWeek> TrainingWeeks { get; set; }
  }
}

File: GymNotes.Angular/ClientApp/src/app/Shared/Models/Training/TrainingPlan.ts
import { TrainingWeek } from './TrainingWeek';
import { User } from '../User';

export interface TrainingPlan {
	name: string;
	description: string;
	trainingWeeks: TrainingWeek[];
	owner: User;
	creator: User;
	modifiedAt: Date;
	id: number;
	isMain: boolean;
	isFavorite: boolean;
}
